robots/src/robot.cpp

```
@@ -13,24 +15,53 @@
+}
+
+Robot* GetRobotByName(const char* name) {
+  if (YardForceRobot::BoardIsCompatible() && strncmp(name, YardForceRobot::FirmwareName(), 50) == 0)
+    return new YardForceRobot{};
+  if (YardForce_V4Robot::BoardIsCompatible() && strncmp(name, YardForce_V4Robot::FirmwareName(), 50) == 0)
+    return new YardForce_V4Robot{};
+  if (Universal5SRobot::BoardIsCompatible() && strncmp(name, Universal5SRobot::FirmwareName(), 50) == 0)
+    return new Universal5SRobot{};
+  if (Universal7SRobot::BoardIsCompatible() && strncmp(name, Universal7SRobot::FirmwareName(), 50) == 0)
+    return new Universal7SRobot{};
+  if (WorxRobot::BoardIsCompatible() && strncmp(name, WorxRobot::FirmwareName(), 50) == 0) return new WorxRobot{};
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If CC_SECTION is a macro then please configure it.

(unknownMacro)